### PR TITLE
Add customized rendering

### PR DESCRIPTION
**Pull request by guillep2k**
- [x] Moved semver sorting to cache; it doesn't make sense to sort every time the directory is served (and I needed to move it, so...).
- [x] Added customizable rendering for folders: icon and description depending on a json file present in each folder (`.s3-customize.json`).
- [x] Added column "Description"
- [x] Added "pre-rendered" directories and folders (i.e. sorted and decorated according to customization).
- [x] Moved drone test files to `test/` directory to prevent accidental commit (or accidentally deleting of actual WIP 😭).
- [x] Added icons "current", "release", "working", "older", meaning respectively "current release", "older release", "a working branch for current release", "a working branch for an older release".
- [x] Get better graphics for the proposed icons
- [x] Fix template for better rendering (it's currently ugly)

Attached a rendered version of the test site.

### UPDATED:

[rendered.txt](https://github.com/techknowlogick/caddy-s3browser/files/4708246/rendered.txt)

### CODE REVIEW

File: template.go [hunk 1]
```diff
@@ -336,34 +292,37 @@ footer {
...
 					{{- end}}
-					{{ range $name := SemSort $ .Dir.Folders }}
+					{{ range $dir := .Dir.RenderedDirs }}
 						<tr class="file">
 							<td></td>
 							<td>
-								<a href="{{ html (PathJoin $.Dir.Path $name) }}">
-									<svg width="1.5em" height="1em" version="1.1" viewBox="0 0 317 259"><use xlink:href="#folder"></use></svg>
-									<span class="name">{{ html $name }}</span>
+								<a href="{{ html (PathJoin $.Dir.Path $dir.Name) }}">
```
Comment (techknowlogick) [inline template.go]:
```suggestion
								<a href="{{ html (PathJoin $.Dir.Path $dir.Name) }}"{{ if ne $dir.Description "" }} title="{{ $dir.Description }}{{- end}}">
```
This gives a title attribute for better accessibility.

**Comment by techknowlogick** [comment]
For the template I am fine with how it is as it is aligned with the browse template built into caddy itself, and for the specific icons used for accessibility sake it is not good to distinguish something by colour alone, and so I liked the idea you had for dev/RC where there were something like gears on the folder.

I'm a fan of 1. moving generated files for testing to be ignored by git (so many times I stage tested files for commit only to realize last minute before committing), 2. Only sorting the one time vs, every time the templates are rendered, 3. The advanced customizations possible via user provided config, and 4. the distinct icons per type (although as mentioned above maybe something other than just colour could be used). 

I have the one nit, as it is something that can improve accessibility, but other wise I think this can be merged soon. Thanks again for all your PRs 😄 👏

**Comment by guillep2k** [comment]
Great! So I'd only need to polish the output file (which I left raw by the way). Colors were just a test for "proof of concept", they are awful!! 😆 But I wanted to get your opinion on the _golang_ changes before proceeding with the UI part. I'll get into it, then. I'll have to do some changes because now there's a contention between the width of the version column and the with of the description's. Also I'll use the gear icon as advertised. 😁

**Comment by guillep2k** [comment]
I've just finished the template updates. I hope you like the new icons. If you'd rather use the old icons, I have no problem, but I'd like to use some way of highlighting the current release (e.g. applying **bold** to the release description, a different background color, etc.). Just let me know your preference.

I've updated the rendered sample at the first comment (`rendered.txt`). Here's a capture of how it would look for Gitea as of now:

image

**Comment by guillep2k** [comment]
Eventually I could add the ability to specify an external stylesheet from the same folder.

**Comment by guillep2k** [comment]
Added a slightly greenish background for the current release. Updated the samples.

**Review by techknowlogick — APPROVED** [review]

**Comment by guillep2k** [comment]
Sorry, I forgot about the `title` attribute. 😅

**Pull request merged**